core/src/main/java/com/netflix/conductor/core/utils/ExternalPayloadStorageUtils.java

```diff
@@ -192,6 +192,8 @@ public <T> void verifyAndUpload(T entity, PayloadType payloadType) {
             }
         } catch (TransientException te) {
             throw te;
+        } catch (TerminateWorkflowException twe) {
```

Comment:
I presume that this fails the workflow, but the task whose payload is large will not be marked as FAILED. Should we try to fail the respective task?

Reply:
It will fail the task as well if the terminateworkflowexception is due to task exceed size limit: https://github.com/Netflix/conductor/blob/ecfa7b10e620b6edc230d10d3eeed932e9f04784/core/src/main/java/com/netflix/conductor/core/utils/ExternalPayloadStorageUtils.java#L220

There is a test case added verified that as well